bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapOneCallHistoryHandler.java

```diff
@@ -0,0 +1,323 @@
+            State state = UnDefType.UNDEF;
+            switch (channelId) {
+                case CHANNEL_TIME_STAMP:
+                    state = getDateTimeTypeState(historyData.getDt());
+                    break;
+                case CHANNEL_CONDITION:
+                    if (!historyData.getWeather().isEmpty()) {
+                        state = getStringTypeState(historyData.getWeather().get(0).getDescription());
+                    }
+                    break;
+                case CHANNEL_CONDITION_ID:
+                    if (!historyData.getWeather().isEmpty()) {
+                        state = getStringTypeState(Integer.toString(historyData.getWeather().get(0).getId()));
+                    }
+                    break;
+                case CHANNEL_CONDITION_ICON:
+                    if (!historyData.getWeather().isEmpty()) {
+                        state = getRawTypeState(
+                                OpenWeatherMapConnection.getWeatherIcon(historyData.getWeather().get(0).getIcon()));
+                    }
+                    break;
+                case CHANNEL_CONDITION_ICON_ID:
+                    if (!historyData.getWeather().isEmpty()) {
+                        state = getStringTypeState(historyData.getWeather().get(0).getIcon());
+                    }
+                    break;
+                case CHANNEL_TEMPERATURE:
+                    state = getQuantityTypeState(historyData.getTemp(), CELSIUS);
+                    break;
+                case CHANNEL_APPARENT_TEMPERATURE:
+                    state = getQuantityTypeState(historyData.getFeelsLike(), CELSIUS);
+                    break;
+                case CHANNEL_PRESSURE:
+                    state = getQuantityTypeState(historyData.getPressure(), HECTO(PASCAL));
+                    break;
+                case CHANNEL_HUMIDITY:
+                    state = getQuantityTypeState(historyData.getHumidity(), PERCENT);
+                    break;
+                case CHANNEL_DEW_POINT:
+                    state = getQuantityTypeState(historyData.getDewPoint(), CELSIUS);
+                    break;
+                case CHANNEL_WIND_SPEED:
+                    state = getQuantityTypeState(historyData.getWindSpeed(), METRE_PER_SECOND);
+                    break;
+                case CHANNEL_WIND_DIRECTION:
+                    state = getQuantityTypeState(historyData.getWindDeg(), DEGREE_ANGLE);
+                    break;
+                case CHANNEL_GUST_SPEED:
+                    state = getQuantityTypeState(historyData.getWindGust(), METRE_PER_SECOND);
+                    break;
+                case CHANNEL_CLOUDINESS:
+                    state = getQuantityTypeState(historyData.getClouds(), PERCENT);
+                    break;
+                case CHANNEL_VISIBILITY:
+                    @Nullable
+                    State tempstate = new QuantityType<>(localWeatherData.getCurrent().getVisibility(), METRE)
```

Comment:
Why do you access `localWeatherData` when updating channel group hourly history`Is this by reason or a mistake?

Reply:
Mistake. Corrected